shared/crypto/slip.go

```diff
@@ -0,0 +1,156 @@
+package crypto
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha512"
+	"encoding/binary"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	// PoktAccountPathFormat used for HD key derivation where 635 is the SLIP-0044 coin type
+	// To be used with fmt.Sprintf to generate child keys
+	PoktAccountPathFormat = "m/44'/635'/%d'"
+	// firstHardenedKey is the index for the first hardened key for ed25519 keys
+	firstHardenedKey = uint32(0x80000000)
+	maxHardenedKey   = ^uint32(0)
+	MaxChildKeyIndex = maxHardenedKey - firstHardenedKey
+	// As defined in: https://github.com/satoshilabs/slips/blob/master/slip-0010.md#master-key-generation
+	seedModifier = "ed25519 seed"
```

Comment:
s/seedModifier/hmacKey

Reply:
@Olshansk I think this should stay as `seedModifier` it is only the first part of the HMAC key and the seed provided (which in theory should be a mnemonic or something like this or just a key's seed) is the other part